### PR TITLE
Connect to api Pull request

### PR DESCRIPTION
**Pull request by HERMON-1995**
**In this project, I connected my React & Redux application to an existing [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) to create and remove books from a server.** Here are some of the changes: 

- Used [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make your HTTP requests to the backend.
- Fetched  books from the Bookstore API after the component has rendered
- Refactored the add book and remove book features to persist changes in the server

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -1,36 +1,46 @@
...
+  await fetch(URL, {
+    method: 'POST',
+    body: new URLSearchParams({
+      item_id: uuidv4(),
+      author: book.author,
+      title: book.title,
+      category: book.category,
+    }),
+  });
+  const res = await fetch(URL);
```
Comment (Olamarx) [inline src/redux/books/booksSlice.js]: - [ ] It is required to make all the requests with axios. Kindly consider changing fetch and make use of axios as you have used in the api-content file.

File: src/redux/books/booksSlice.js [hunk 2]
```diff
@@ -1,36 +1,46 @@
...
+  const json = await res.json();
+  return json;
 });
 
-export const { addBook, removeBook } = booksSlice.actions;
+export const removeBook = createAsyncThunk('bookstore/books/REMOVE', async (bookId) => {
+  await fetch(`${URL}/${bookId}`, {
```
Comment (Olamarx) [inline src/redux/books/booksSlice.js]: - [ ] Same thing here as regards axios.

File: src/redux/books/booksSlice.js [hunk 3]
```diff
@@ -1,36 +1,46 @@
...
+    method: 'DELETE',
+  });
+  const res = await fetch(URL);
```
Comment (Olamarx) [inline src/redux/books/booksSlice.js]: - [ ] Same thing here as regards axios.

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @HERMON-1995, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters passed ✔️ 
- Readme file professional ✔️ 
- Redux toolkit ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js (hunk 3 above)
Comment (HERMON-1995) [reply src/redux/books/booksSlice.js]: Thank you will change it right away.😊

**Review by gemmen29 — APPROVED** [review]
Hi @HERMON-1995, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by HERMON-1995** [comment]
Thank you for approving my project! 😄

**Pull request merged**